src/psfmachine/superstamp.py

```diff
@@ -0,0 +1,533 @@
+        prior_sigma = (
+            np.ones(self.mean_model.shape[0])
+            * 5
+            * np.abs(self.source_flux_estimates) ** 0.5
+        )
+        self.model_flux_frame = np.zeros(self.flux.shape) * np.nan
+        self.ws_frame = np.zeros((self.nt, self.nsources))
+        self.werrs_frame = np.zeros((self.nt, self.nsources))
+        f = self.flux
+        fe = self.flux_err
+        for tdx in tqdm(
+            range(self.nt),
+            desc=f"Fitting {self.nsources} Sources (per frame model)",
+            disable=self.quiet,
+        ):
+            X = self.mean_model_frame[tdx].copy().T
+            sigma_w_inv = X.T.dot(X.multiply(1 / fe[tdx][:, None] ** 2)).toarray()
+            sigma_w_inv += np.diag(1 / (prior_sigma ** 2))
+            B = X.T.dot((f[tdx] / fe[tdx] ** 2))
+            B += prior_mu / (prior_sigma ** 2)
+            self.ws_frame[tdx] = np.linalg.solve(sigma_w_inv, np.nan_to_num(B))
+            self.werrs_frame[tdx] = np.linalg.inv(sigma_w_inv).diagonal() ** 0.5
+            self.model_flux_frame[tdx] = X.dot(self.ws_frame[tdx])
+            nodata = np.asarray(self.source_mask.sum(axis=1))[:, 0] == 0
+            # These sources are poorly estimated
+            nodata |= (self.mean_model_frame[tdx].max(axis=1) > 1).toarray()[:, 0]
+            self.ws_frame[tdx, nodata] *= np.nan
+            self.werrs_frame[tdx, nodata] *= np.nan
+
+    @staticmethod
+    def from_file(fname, magnitude_limit=18, dr=2, **kwargs):
+        """
+        Reads data from files and initiates a new FFIMachine class.
+        Parameters
+        ----------
+        fname : string or list of strings
+            Name of the FITS files to be parsed.
+        dr : int
+            Gaia data release to be use, default is 2, options are DR2 and EDR3.
+        **kwargs : dictionary
+            Keyword arguments that defines shape model in a `Machine` object.
+        Returns
+        -------
+        FFIMachine : Machine object
+            A Machine class object built from the FFI.
+        """
+        (
+            wcs,
+            time,
+            flux,
+            flux_err,
+            ra,
+            dec,
+            column,
+            row,
+            metadata,
+        ) = _load_file(fname)
+
+        sources = _get_sources(
+            ra,
+            dec,
+            wcs,
+            magnitude_limit=magnitude_limit,
+            epoch=time.jyear.mean(),
+            ngrid=(2, 2) if flux.shape[1] <= 500 else (5, 5),
+            dr=dr,
+            img_limits=[[row.min(), row.max()], [column.min(), column.max()]],
+        )
+
+        return SSMachine(
+            time.jd,
+            flux,
+            flux_err,
+            ra.ravel(),
+            dec.ravel(),
+            sources,
+            column.ravel(),
+            row.ravel(),
+            wcs=wcs,
+            meta=metadata,
+            **kwargs,
+        )
+
+    def fit_lightcurves(
```

Comment:
Is this not an inherited method from the subclass?

Reply:
Not quite, this version has the option to do:
- mean-PSF photometry
- mean-PSF photometry with VA
- cadence-PSF photometry
- SAP

Reply:
I do want to move all the code that creates the LC metadata out of `tpf.py` and `superstamp.py` to a dedicated function (maybe in `utils.py`) to remove repeated code. But I want to do that in a separate PR along with more code cleaning I want to do. 
What do you think?

Reply:
I think this sounds like a great idea!